specs/agents/tracing-spans-handling-huge-traces.md

```diff
@@ -0,0 +1,17 @@
+# Handling huge traces
+
+Instrumenting applications that make lots of requests (such as 10k+) to backends like caches or databases can lead to several issues:
+- A significant performance impact in the target application.
+  For example due to high allocation rate, network traffic, garbage collection, additional CPU cycles for serializing, compressing and sending spans, etc.
+- Dropping of events in agents or APM Server due to exhausted queues.
+- High load on the APM Server.
+- High storage costs.
+- Decreased performance of the Elastic APM UI due to slow searches and rendering of huge traces.
+- Loss of clarity and overview (--> decreased user experience) in the UI when analyzing the traces.
+
+Agents can implement several strategies to mitigate these issues:
```

Comment:
Can we add some pseudo code / algorithm demonstrating how all of these strategies / parts would ideally work in combination? Something like this:


On span end or composite span end:

```java
if (span count on current transaction > TRANSACTION_MAX_SPANS) {
   ... drop span
   ... collect statistics for dropped spans
} else {
   if (compression possible) {
      ... apply compression
   } else if (ended span duration < SPAN_MIN_DURATION OR ended _composite_ span duration < SPAN_MIN_DURATION) {
      ... drop ended span
      ... collect statistics for dropped spans
   }
}
```